### PR TITLE
Add rushstack-based eslint config

### PR DESCRIPTION
**Pull request by tylerbutler**
This will enable further investigation of #3921.

### CODE REVIEW

File: common/build/eslint-config-fluid/rushstack.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                "@typescript-eslint/no-non-null-assertion": "error",
+                "@typescript-eslint/no-unnecessary-type-assertion": "error",
+                "max-len": [
+                    "error",
+                    {
+                        "ignoreRegExpLiterals": false,
+                        "ignoreStrings": false,
+                        "code": 120
+                    }
+                ],
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/rushstack.js:93]: These don't appear to be enabled in the rushstack config.

File: common/build/eslint-config-fluid/rushstack.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+                // DISABLED INTENTIONALLY
+                "@rushstack/typedef-var": "off",
+                "@typescript-eslint/explicit-function-return-type": "off",
+                "@typescript-eslint/no-explicit-any": "off",
+                "@typescript-eslint/member-ordering": "off",
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/rushstack.js:99]: These are disabled because they're noisy in the packages I spot checked.

Comment (DLehenbauer) [reply common/build/eslint-config-fluid/rushstack.js:99]: Yeah, those rules are fairly opinionated / pedantic.

File: common/build/eslint-config-fluid/rushstack.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// This is a workaround for https://github.com/eslint/eslint/issues/3458
+require("@rushstack/eslint-config/patch/modern-module-resolution");
```
Comment (DLehenbauer) [inline common/build/eslint-config-fluid/rushstack.js:7]: I was wondering why I didn't see a new dependency for `@rushstack/eslint-config`, but I bet you already have one for "modern-module-resolution"...

**Review by DLehenbauer — APPROVED** [review]

**Pull request merged**